sources/updater/update_metadata/src/lib.rs

```diff
@@ -296,58 +299,86 @@ impl Manifest {
 impl Update {
     /// Returns the update wave that Updog belongs to, based on the seed value.
     /// Depending on the waves described in the update, the possible results are
-    /// - Some wave described by a start and end time.
-    /// - The "0th" wave, which has an "end" time but no specified start time.
-    /// - The last wave, which has a start time but no specified end time.
+    /// - Some wave described by a start and end time, and the starting seed and ending seed.
+    /// - The "0th" wave, which has an "end" time but no specified start time, and the starting seed.
+    /// - The last wave, which has a start time but no specified end time, and the starting seed.
     /// - Nothing, if no waves are configured.
     pub fn update_wave(&self, seed: u32) -> Option<Wave> {
-        let start = self
-            .waves
-            .range((Included(0), Excluded(seed)))
-            .last()
-            .map(|(_, wave)| *wave);
-        let end = self
+        let start_wave = self.waves.range((Included(0), Excluded(seed))).last();
+        let end_wave = self
             .waves
             .range((Included(seed), Included(MAX_SEED)))
-            .next()
-            .map(|(_, wave)| *wave);
-
-        match (start, end) {
-            (None, Some(end)) => Some(Wave::Initial { end }),
-            (Some(start), Some(end)) => Some(Wave::General { start, end }),
-            (Some(start), None) => Some(Wave::Last { start }),
+            .next();
+
+        match (start_wave, end_wave) {
+            // Note that the key for each wave entry is the starting seed for that wave, the value is the DateTime
+            (None, Some(end_wave)) => Some(Wave::Initial {
+                end_time: *end_wave.1,
+                end_seed: *end_wave.0,
+            }),
```

Comment:
I'm hoping we can bring together the two wave representations at some point, but for now, can we use pattern matching instead of tuple indexing?  Something like this, and similar below...
```suggestion
            (None, Some((end_seed, end_time))) => Some(Wave::Initial { end_seed, end_time }),
```